Update Signum

JesusMcCloud opened the pull request:
this one includes some kotest workarounds, which should be resolved once a new kotest release is out. most notably the kotest config variable and using our own freespec.

Closes #347

JesusMcCloud commented on vck-openid-ktor/src/commonTest/kotlin/KotestConfig.kt:
```diff
@@ -11,4 +11,5 @@ class KotestConfig : AbstractProjectConfig() {
-}
\ No newline at end of file
+}
+val inited = KotestConfig()
```
workaround for current Kotest bugs. I should be able to add a KSP-powered workaround that does away with this but since its "just" some wonky-looking code in the tests i figured it should not be a blocker. If Sam is keeping busy in the right way there might not even be any need to workarounds by next week. i'd keep this an open an issue to revisit it in one week's time.

nodh replied:
Shouldn't it be called `initialized` or `isInitialized`? Or something with workaround in the name

JesusMcCloud commented on vck-openid-ktor/src/commonTest/kotlin/at/asitplus/wallet/lib/ktor/openid/OpenId4VciClientTest.kt:
```diff
@@ -58,14 +60,18 @@ import io.kotest.matchers.types.shouldBeInstanceOf
-
-class OpenId4VciClientTest : FunSpec() {
+class OpenId4VciClientTest : FreeSpec() {
+val init= inited
```
this accompanies the workaround from above

nodh replied:
Then please at least intend it the right way.
Also, is there no more `FunSpec`?

JesusMcCloud replied:
It's a custom FreeSpec to work around missing test report generation. Hopefully the next kotest release wires everything correctly out of the box then we won't need such hacks any more

JesusMcCloud commented:
@nodh want me to rebase before you review again or after?

nodh reviewed (changes requested):
Please rebase, and add a changelog entry

JesusMcCloud commented:
something's dodgy in vck-openid-ktor: it's not correctly building for android and still using JDK 17 instead of 1.8 for the android target (because no properly setup, dedicated android target seems to be present). i need to investigate further. can someone please try a composite build with this state and https://github.com/a-sit-plus/valera/pull/375 and report back

JesusMcCloud commented on vck-openid-ktor/src/commonTest/kotlin/KotestConfig.kt:
```diff
@@ -1,9 +1,14 @@
+package io.kotest.provided
```
Here's my rationale behind it:
* This is required global config for all tests in a module, so it should be located in the common test sources root
* The package name and class name must be fixed for Kotest to pick it up

same applies for all other comments related to kotest config files. The other tests I'll move

JesusMcCloud commented:
I addressed all comments and rebased. I assume you want to merge, given you self-assigned? I still have no clue what's going on with the botched Android target for the openid-ktor module…
